src/cmds/qsub_sup.c

```diff
@@ -1001,6 +1132,85 @@ writer(int s)
 	return;
 }
 
+/**
+ * @brief
+ * 	Packet Writer process: reads from stdin, and writes
+ * 	data out to the rem socket
+ *
+ * @param[in] s - file descriptor
+ *
+ * @return Void
+ *
+ */
+void
+pkt_writer(int s)
```

Comment:
there's a huge overlap between this function and writer(). The only difference is transport_send_pkt() vs CS_write(). Can you unify the two functions in one, with an extra option to signify if we're sending a packet or not? 

In fact an additional option might not be required, since you test if the the socket is ready for encryption directly inside the function..

Similarly with reader/pkt_reader and reader_Xjob/pkt_reader_Xjob